tests/apollo/util/bft.py

```diff
@@ -75,7 +75,10 @@ def with_trio(async_fn):
     """ Decorator for running a coroutine (async_fn) with trio. """
     @wraps(async_fn)
     def trio_wrapper(*args, **kwargs):
-        return trio.run(async_fn, *args, **kwargs)
+        if "bft_network" in kwargs:
```

Comment:
Actually, we should have a separate parameter for disabling the @with_trio decorator (say, "already_in_trio=True/False")